torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1521,18 +1521,19 @@ int listAdd(Stack& stack) {
   pop(stack, a, b);
 
   c10::List<T> ret;
-  const auto total_size = a.size() + b.size();
-  ret.reserve(total_size);
-  for (T a_element : a) {
-    ret.push_back(std::move(a_element));
-  }
-  for (T b_element : b) {
-    ret.push_back(std::move(b_element));
+
+  if (a.use_count() == 1) {
```

Comment:
Wondering if we should have a `List::make_unshared_copy()` that ensures that after calling it, `use_count == 1` and does a copy if it needs to, or something for this. Feels like this could be useful in other ops as well.